src/code/z_player_lib.c

```diff
@@ -1,35 +1,132 @@
-} BowStringData; // size = 0x10
+    /* 0x00 */ s16 walkSpeedMax;
+    /* 0x02 */ s16 sideWalkAnimSpeedBase;
+    /* 0x04 */ s16 sideWalkAnimSpeed;
+    /* 0x06 */ s16 unk_06;
+    /* 0x08 */ s16 walkAnimSpeedBase;
+    /* 0x0A */ s16 walkAnimSpeed;
+    /* 0x0C */ s16 runAnimSpeedBase;
+    /* 0x0E */ s16 runAnimSpeed;
+    /* 0x10 */ s16 unk_10;
+    /* 0x12 */ s16 runSpeedMax;
+    /* 0x14 */ s16 gravity;
+    /* 0x16 */ s16 unk_16;           // jump related
+    /* 0x18 */ s16 highJumpSpeedMin; // minimum speed to trigger a high jump
+    /* 0x1A */ s16 highJumpSpeed;
+    /* 0x1C */ s16 highJumpSpeedBase;
+    /* 0x1E */ s16 smallJumpSpeed;
+    /* 0x20 */ s16 unk_20;
+} PlayerMovementBehavior; // size = 0x22
+
+// Info about how the player should walk, run, and jump under various conditions
+static PlayerMovementBehavior sMovementInfo[PLAYER_BOOTS_MAX] = {
+    { 200, 1000, 300, 700, 550, 270, 600, 350, 800, 600, -100, 600, 590, 750, 125, 200, 130 }, // Normal (adult)
+    { 200, 1000, 300, 700, 550, 270, 1000, 0, 800, 300, -160, 600, 590, 750, 125, 200, 130 },  // Iron
+    { 200, 1000, 300, 700, 550, 270, 600, 600, 800, 550, -100, 600, 540, 270, 25, 0, 130 },    // Hover
+    { 200, 1000, 300, 700, 380, 400, 0, 300, 800, 500, -100, 600, 590, 750, 125, 200, 130 },   // Indoor
+    { 80, 800, 150, 700, 480, 270, 600, 50, 800, 550, -40, 400, 540, 270, 25, 0, 80 },         // Iron underwater
+    { 200, 1000, 300, 800, 500, 400, 800, 400, 800, 550, -100, 600, 540, 750, 125, 400, 200 }, // Normal (child)
+};
 
-FlexSkeletonHeader* gPlayerSkelHeaders[] = { 0x060377F4, &gLinkChildSkel };
+void Player_SetMovementBehavior(GlobalContext* globalCtx, Player* this) {
+    s32 currentBoots;
+    PlayerMovementBehavior* entry;
 
-s16 sBootData[PLAYER_BOOTS_MAX][17] = {
-    { 200, 1000, 300, 700, 550, 270, 600, 350, 800, 600, -100, 600, 590, 750, 125, 200, 130 },
-    { 200, 1000, 300, 700, 550, 270, 1000, 0, 800, 300, -160, 600, 590, 750, 125, 200, 130 },
-    { 200, 1000, 300, 700, 550, 270, 600, 600, 800, 550, -100, 600, 540, 270, 25, 0, 130 },
-    { 200, 1000, 300, 700, 380, 400, 0, 300, 800, 500, -100, 600, 590, 750, 125, 200, 130 },
-    { 80, 800, 150, 700, 480, 270, 600, 50, 800, 550, -40, 400, 540, 270, 25, 0, 80 },
-    { 200, 1000, 300, 800, 500, 400, 800, 400, 800, 550, -100, 600, 540, 750, 125, 400, 200 },
-};
+    REG(27) = 2000;
+    REG(48) = 370;
+
+    currentBoots = this->currentBoots;
+    if (currentBoots == PLAYER_BOOTS_KOKIRI) {
+        if (LINK_IS_CHILD) {
+            currentBoots = PLAYER_BOOTS_EX_KOKIRI_CHILD;
+        }
+    } else if (currentBoots == PLAYER_BOOTS_IRON) {
+        if (this->stateFlags1 & 0x8000000) {
+            currentBoots = PLAYER_BOOTS_EX_IRON_UNDERWATER;
+        }
+        REG(27) = 500;
+        REG(48) = 100;
+    }
+
+    entry = &sMovementInfo[currentBoots];
+    REG(19) = entry->walkSpeedMax;
+    REG(30) = entry->sideWalkAnimSpeedBase;
+    REG(32) = entry->sideWalkAnimSpeed;
```

Comment:
what does base mean in this context? how does it differ from the normal speed?

Reply:
I did this like 4-5 months ago so don't remember tbh, this stuff was difficult to name but I think the "base" speed was kind of like the minimum speed which they gets multiplied

Reply:
it might be worth to revisit and confirm your reasoning, and potentially rename if it seems fitting. base doesnt really make sense to me at first glance without reading some more code 
i understand it being hard to name tho yeah, it seems like it